xbmc/video/VideoDatabase.cpp

```diff
@@ -4290,7 +4290,8 @@ CVideoInfoTag CVideoDatabase::GetDetailsForMovie(const dbiplus::sql_record* cons
 
   if (getDetails)
   {
-    GetCast(details.m_iDbId, MediaTypeMovie, details.m_cast);
+    if (getDetails & VideoDbDetailsCast)
```

Comment:
Shouldnt be a need to test getDetails, as you are already in an if block that does that check, therefore it will always be true here.